### PR TITLE
chore(deps): upgrade dependencies

### PR DESCRIPTION
**Pull request by 2eha0**
### Summary

* upgrade vite and its relative dependencies
* upgrade vue and its relative dependencies
* upgrade node to LTS (v18.20.4)
* upgrade typescript
* upgrade eslint
* remove unused dependencies
* dedupe dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,14 +42,13 @@
     "@kong-ui-public/misc-widgets": "^2.1.16",
     "@kong/icons": "^1.14.0",
     "@kong/kongponents": "^9.5.1",
-    "@material-design-icons/font": "^0.14.9",
-    "axios": "^1.7.4",
-    "dayjs": "^1.11.10",
-    "marked": "^5.1.0",
```
Comment (2eha0) [inline package.json:48]: unused

File: package.json [hunk 2]
```diff
@@ -69,18 +68,19 @@
     "husky": "^8.0.3",
     "lint-staged": "^13.2.2",
     "lodash-es": "^4.17.21",
-    "postcss-html": "^1.5.0",
-    "rollup-plugin-visualizer": "^5.9.2",
-    "sass": "^1.63.4",
+    "postcss": "^8.4.44",
```
Comment (2eha0) [inline package.json:71]: unify postcss version by manually install

**Review by nekolab — APPROVED** [review]
LGTM

**Review by Leopoldthecoder — APPROVED** [review]
LGTM

**Pull request merged**